sync/syncer.go

```diff
@@ -194,28 +196,48 @@ func (s *Syncer) run() {
 }
 
 //fires a sync every sm.syncInterval or on force space from outside
-func NewSync(srv service.Service, layers *mesh.Mesh, txpool TxMemPool, atxpool AtxMemPool, tv TxValidator, bv BlockValidator, poetdb PoetDb, conf Configuration, clockSub clockSubscriber, currentLayer types.LayerID, logger log.Log) *Syncer {
+func NewSync(srv service.Service, layers *mesh.Mesh, txpool TxMemPool, atxpool AtxMemPool, tv TxValidator, bv BlockValidator, poetdb PoetDb, conf Configuration, clockSub clockSubscriber, currentLayer types.LayerID, atxsLimit int, logger log.Log) *Syncer {
+	fastValidator := func(block *types.Block) error {
+
+		if len(block.AtxIds) > atxsLimit {
+			logger.Error("Too many atxs in block expected<=%v actual=%v", atxsLimit, len(block.AtxIds))
+			return errTooManyAtxs
+		}
+
+		// validate unique tx atx
+		if err := validateUniqueTxAtx(block); err != nil {
+			return err
+		}
+
+		// block eligibility
+		if eligible, err := bv.BlockSignedAndEligible(block); err != nil || !eligible {
```

Comment:
I would check this first: it is much easier to create a (non-valid) block with unique txs than a block with a valid signature

Reply:
Done